greeter/README.md

```diff
@@ -1,21 +1,706 @@
+
+```ts
+/**
+ * Greeting message response
+ */
+export interface Message {
+  timestamp: Date;
+  language: string;
+  greeting: string;
+}
+```
+
+### Step 3: Create a Greeting Controller
+
+1. Create an endpoint `/greet/{name}` that takes a name and returns a greeting message in the language that is specified in the HTTP request. Run the Controller generator:
+
+   ```sh
+   $ lb4 controller
+   ? Controller class name: Greeting
+   Controller Greeting will be created in src/controllers/greeting.controller.ts
+
+   ? What kind of controller would you like to generate? Empty Controller
+   create src/controllers/greeting.controller.ts
+   update src/controllers/index.ts
+
+   Controller Greeting was created in src/controllers/
+   ```
+
+   Then add these lines as a mock service to the controller
+
+   ```ts
+   // Will be replaced by the greeting service from extension later
+   export class SimpleGreetingService extends GreetingService {
+     async greet(language: string, name: string): Promise<string> {
+       return `${name} in language ${language}`;
+     }
+   }
+   ```
+
+2. In the generated GreetingController in `src/controllers/greeting.controller.ts`, modify the constructor:
+
+The `@inject` decorator here is a usage of Context and IoC, it binds components to your app. And a binding connects its value to a unique key as the address to access the entry in a context. Such injections/bindings are useful in real world application design -- in our case, instead of having the controller handled all logics, Context and IoC decouple the execution and make the app greater modularity with components.
+
+```ts
+class GreetingController {
+  constructor(
+    @inject(GREETING_SERVICE) private greetingService: GreetingService,
+    @inject(RestBindings.Http.REQUEST) private request: Request,
+  ) {}
+  // define @post, @get here..
+}
+```
+
+3. Add the endpoint `/greet/{name}` which calls the [GreetingService](https://github.com/strongloop/loopback-next/blob/master/examples/greeter-extension/src/greeting-service.ts) in the greeter-extension that we've created already. It accepts language English or Chinese. If other language is specified, it will fall back to English.
+
+   ```ts
+   @get('/greet/{name}', {
+   responses: {
+   '200': {
+       description: '',
+       content: {
+       'application/json': {
+           schema: {
+           type: 'object',
+           properties: {
+               timestamp: 'string',
+               language: 'string',
+               message: 'string',
+           },
+           },
+       },
+       },
+   },
+   },
+   })
+   async greet(
+       @param.path.string('name') name: string,
+       @param.header.string('Accept-Language') lan: string,
+   ): Promise<Message> {
+       const language: string = this.request.acceptsLanguages(['en', 'zh']) || 'en';
```

Comment:
> const language: string = this.request.acceptsLanguages(['en', 'zh']) || 'en'; 

This is super annoying/confusing and causes the application to fail to show  a french return value even after we add the french greeter.

Let's use:

const language: string = lan;